pkg/util/ipvs/testing/fake_test.go

```diff
@@ -113,46 +113,59 @@ func TestRealServer(t *testing.T) {
 		Port:     uint16(80),
 		Protocol: string("TCP"),
 	}
+	rss := []*utilipvs.RealServer{
+		{net.ParseIP("172.16.2.1"), 8080, 1},
+		{net.ParseIP("172.16.2.2"), 8080, 2},
+		{net.ParseIP("172.16.2.3"), 8080, 3},
+	}
 	err := fake.AddVirtualServer(vs)
 	if err != nil {
 		t.Errorf("Fail to add virutal server, error: %v", err)
 	}
-	// Add a real server to the virtual server
-	rs1 := &utilipvs.RealServer{
-		Address: net.ParseIP("172.16.2.1"),
-		Port:    uint16(8080),
-		Weight:  1,
-	}
-	err = fake.AddRealServer(vs, rs1)
-	if err != nil {
-		t.Errorf("Fail to add real server, error: %v", err)
-	}
-	// Add another real server to the virtual server
-	rs2 := &utilipvs.RealServer{
-		Address: net.ParseIP("172.16.3.2"),
-		Port:    uint16(8080),
-		Weight:  2,
-	}
-	err = fake.AddRealServer(vs, rs2)
-	if err != nil {
-		t.Errorf("Fail to add real server, error: %v", err)
-	}
-	// List all real servers of the virtual server
-	list, err := fake.GetRealServers(vs)
-	if err != nil {
-		t.Errorf("Fail to get real servers of the virtual server, error: %v", err)
-	}
-	if len(list) != 2 {
-		t.Errorf("Expect 2 virutal servers, got: %d", len(list))
+	// Add real server to the virtual server
+	for i := range rss {
+		if err = fake.AddRealServer(vs, rss[i]); err != nil {
+			t.Errorf("Fail to add real server, error: %v", err)
+		}
 	}
 	// Delete a real server of the virtual server
-	err = fake.DeleteRealServer(vs, rs2)
-	list, err = fake.GetRealServers(vs)
-	if err != nil {
-		t.Errorf("Fail to get real servers of the virtual server, error: %v", err)
+	// Make sure any position of the list can be real deleted
+	rssLen := len(rss)
+	for i := range rss {
```

Comment:
Still too complicated...

You fill the deleted virtual server back after deleting just because you have trouble in the for loop?

I would suggest:

1. copy rss to a new slice, e.g. rssToDelete

2. 
```
for i := range rssToDelete {
  delete rss[i]
  check the number of elements from rss
}
```

Would it work for you?

Reply:
Actually, I do this also for test delete first/last/middle element in the list.
Cause maybe the function delete an element that not we wanted, then check the number not work. And also have the possible that the special position of the list can't be delete, such as the last one, in the previous case.
But if you think the test is too specific, keep it simple also work for me, I guess.

Reply:
> Cause maybe the function delete an element that not we wanted, then check the number not work. 

I feel like checking the remained number of elements is enough.

Reply:
After all, it's a fake implementation.

Reply:
That's true, let's keep it simpler.